src/commands/hld/reconcile.ts

```diff
@@ -151,15 +151,24 @@ export const reconcileHld = async (
   for (const [serviceRelPath, serviceConfig] of Object.entries(
     managedServices
   )) {
-    const pathBase = path.basename(serviceRelPath);
-    const serviceName = pathBase;
-    logger.info(`Reconciling service: ${pathBase}`);
+    const serviceName =
+      serviceConfig.displayName || path.basename(serviceRelPath);
+
+    // No name, cannot generate proper routes and middlewares
+    if (serviceName === "." || !serviceName) {
+      logger.warn(
```

Comment:
might want to throw an error and exit instead

Reply:
might be a larger design choice as there's a few places where we simply just `continue` instead of failing fast - perhaps we can revisit this